bluepyopt/ephys/mechanisms.py

```diff
@@ -143,6 +147,13 @@ def instantiate_determinism(self, deterministic, icell, isec, sim):
                     'Stoch channel, not %s' %
                     self.suffix)
 
+    def set_stochasticity(self, allow_stochasticity):
+        """Set stochasticity if allowed, else force determinism"""
+        if allow_stochasticity and not self.always_deterministic:
+            self.deterministic = False
+        else:
+            self.deterministic = True
```

Comment:
Would there be a possibility of raising an exception here. If I call set_stochastic on something, I assume the object will be stochastic afterwards, unless if it throws an exception.

Reply:
The function is more of 'whether this should be stochastic or not', than 'set this to stochastic'. The idea is that, for the pre-protocols, we want everything to be deterministic, so we use this method with allow_stochasticity=False, which sets everything to deterministic. Once the preprotocols are done, we want to set back the stochasticity of each mechanism to its original state (True/False), and so we call this function with allow_stochasticity=True. Note that even when stochasticity is allowed, not every mech will be set to non-deterministic.

Reply:
(pre-protocols are protocols such as the ones to search for RMP, Rin, rheobase ..)